src/coreclr/jit/lower.cpp

```diff
@@ -5551,6 +5551,11 @@ GenTree* Lowering::LowerDelegateInvoke(GenTreeCall* call)
     BlockRange().Remove(thisArgNode);
     BlockRange().InsertBefore(call, newThisAddr, newThis, thisArgNode);
 
+    // Keep delegate alive
+    GenTree* baseClone     = comp->gtCloneExpr(base);
+    GenTree* keepBaseAlive = comp->gtNewKeepAliveNode(baseClone);
+    BlockRange().InsertAfter(call, baseClone, keepBaseAlive);
```

Comment:
Is this fix going to introduce CQ regressions? Ideally, the fix for this issue would be zero diff on x86/x64.

We need to keep delegate alive only until we make the call. The delegate does not need to be alive after call.

Reply:
For my knowledge, will all the various stubs that we may go through on the way to the final target automatically keep the right loader allocator without the JIT at minimum putting the delegate instance in a callee-saved register?

Reply:
I think so. There are really only 3:
- Shuffle thunks: hand-emitted assembly so the GC cannot kick in while they are executing
- Multicast stubs: they keep the delegate thisptr alive. (Adding comment that this is important may be nice.)
- Marshalled function pointers: The delegate points to itself, so calling the method on the delegate will keep it alive.

It is possible that I may be missing a corner case somewhere, but I do not expect it to be hard to fix.